interface-spec.md

```diff
@@ -38,9 +43,7 @@ TODO: to what extent should we use typed signatures (`.equals(Node other)`) vers
 
 **Properties:**
 
-- `String .value` blank node name as a string (example: `_:blank3`)
-
-TODO: Does the value always start with an underscore?
+- `String .value` blank node name as a string without leading `_:` (example: `blank3`)
```

Comment:
* Dave Longley <notifications@github.com> [2015-12-18 06:26-0800]
> > @@ -38,9 +43,7 @@ TODO: to what extent should we use typed signatures (`.equals(Node other)`) vers
> >  
> >  **Properties:**
> >  
> > -- `String .value` blank node name as a string (example: `_:blank3`)
> > -
> > -TODO: Does the value always start with an underscore?
> > +- `String .value` blank node name as a string without leading `_:` (example: `blank3`)
> 
> -1
> 
> I don't see the value of removing `_:` from the value -- that is actually part of the identifier for the blank node. Furthermore, blank node identifiers are not part of the RDF abstract syntax -- the presumption that they start with `_:` is only based on the fact that nearly all concrete syntaxes use this notation. While unlikely, it's *possible* that another syntax would not use that form.

So doesn't that make the _: a property of the syntax?

It's true that at present, there's no promise that one can parse that reference the same bnode label and have them unify on that label.
It is quite reasonable that behind closed doors one would want to do that in order to leverage internal identifiers.

  RDF/XML: <rdf:Description rdf:nodeId="proteinProductTP53"><rdfs:label>P53 product</rdfs:label></rdf:Description>
+ Turtle:  _:proteinProductTP53 a :GeneProduct .
= _:proteinProductTP53 a :GeneProduct ; rdfs:label "P53 product" .

That's a bit heretical but quite practical. It also illustrates that the _: is just Turtle's syntax for flagging a bnode.


> If the actual value of a blank node identifier is not going to be found via the `.value` property, then I think there should be a very good argument for taking that approach. IMO, this approach would be unexpected.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/rdfjs/representation-task-force/pull/19/files#r48028993

-- 
-ericP

office: +1.617.599.3509
mobile: +33.6.80.80.35.59

(eric@w3.org)
Feel free to forward this message to any list for any purpose other than
email address distribution.

There are subtle nuances encoded in font variation and clever layout
which can only be seen by printing this message on high-clay paper.